### PR TITLE
First attempt for regexp_tree dictionary type #21694

### PR DESCRIPTION
**Pull request by ivoleg**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Detailed description / Documentation draft:
First attempt to implement RegexpTreeDictionary, with lots of assumptions

This implements #21694.

### CODE REVIEW

File: src/Processors/Formats/Impl/YAMLInputFormat.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    int count = 0;
+    while (!in.eof() && isWhitespaceASCIIOneLine(*in.position()))
+    {
+        ++in.position();
+        ++count;
+    }
+    return count - spaces;
+}
+
+std::pair<String, String> parseString(String & s, ReadBuffer & in)
```
Comment (vdimir) [inline src/Processors/Formats/Impl/YAMLInputFormat.cpp:91]: Is there reasons not to use yaml parser (yaml-cpp)?

File: src/Dictionaries/RegexpTreeDictionary.cpp [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    Block block;
+    while (executor.pull(block))
+    {
+        const auto rows = block.rows();
+	const ColumnPtr id_column_ptr = block.safeGetByPosition(0).column;
+        const ColumnPtr parent_id_column_ptr = block.safeGetByPosition(1).column;
+        const ColumnPtr regexp_column_ptr = block.safeGetByPosition(2).column;
+        const auto attribute_column_ptrs = collections::map<Columns>(
+            collections::range(3, dict_struct.attributes.size()),
+            [&](const size_t attribute_idx) { return block.safeGetByPosition(attribute_idx).column; });
```
Comment (vdimir) [inline src/Dictionaries/RegexpTreeDictionary.cpp:331]: Regular `for` loop would be better here.

File: src/Dictionaries/RegexpTreeDictionary.cpp [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+        callOnDictionaryAttributeType(attribute.type, type_call);
+    }
+    if (update_field_loaded_block)
+        bytes_allocated += update_field_loaded_block->allocatedBytes();
+}
+
+
+void RegexpTreeDictionary::setNodeValue(UInt64 id, UInt64 parent_id, String regexp, std::vector<String> attributes)
+{
+    std::vector<RegexpTreeNode> children;
```
Comment (vdimir) [inline src/Dictionaries/RegexpTreeDictionary.cpp:300]: Seems it's reduntant, not modified anywhere.

File: src/Dictionaries/RegexpTreeDictionary.cpp [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+                }
+                set_value(i, final_attribute);
+                ++keys_found;
+                break;
+            }
+        }
+    }
+    for(size_t i = 0; i < node.regexp.size(); ++i)
+    {
+	regexp_.pop_back();
```
Comment (vdimir) [inline src/Dictionaries/RegexpTreeDictionary.cpp:233]: Maybe `regexp_` should be `vector<string>` ? Actualy it's unclear why we concatenate regexps into this variable

**Review by vdimir — COMMENTED** [review]
Quick look

**Comment by alexey-milovidov** [comment]
I tried to contact @ivoleg with no success.
Unfortunately this PR is abandoned.

**Pull request closed without merging**